controllers/user.js

```diff
@@ -594,6 +685,1369 @@ function logout(req, res) {
+  const userHasSignedUpForRemoveData = hasUserSignedUpForWaitlist(
+    user,
+    "remove_data"
+  );
+
+  let removeData = null; //data broker info
+  let removeAcctInfo = null; //acct info
+
+  if (user.kid) {
+    removeAcctInfo = await getRemoveAcctInfo(user.kid);
+    if (!removeAcctInfo) {
+      console.error(
+        "kid in database, but no account returned from API - likely this account was removed"
+      );
+      const localeError = LocaleUtils.fluentFormat(
+        req.supportedLocales,
+        "remove-error-kid-but-no-acct"
+      );
+      res.render("dashboards", {
+        title: req.fluentFormat("Firefox Monitor"),
+        whichPartial: "dashboards/remove-error",
+        error: localeError,
+      });
+    }
+    removeData = await getRemoveDashData(user.kid);
+
+    if (!show_form && removeData) {
+      removeData.forEach((removeItem) => {
+        removeItem.update_status = FormUtils.convertTimestamp(
+          removeItem.updated_at
+        );
+      });
+    }
+  }
+
+  const experimentFlags = getExperimentFlags(req, EXPERIMENTS_ENABLED);
+
+  let lastAddedEmail = null;
+
+  req.session.user = await DB.setBreachesLastShownNow(user);
+  if (req.session.lastAddedEmail) {
+    lastAddedEmail = req.session.lastAddedEmail;
+    req.session["lastAddedEmail"] = null;
+  }
+
+  let partialString;
+
+  if (show_form) {
+    partialString = "dashboards/remove-form";
+  } else {
+    partialString = "dashboards/remove-dashboard";
+  }
+
+  res.render("dashboards", {
+    title: req.fluentFormat("Firefox Monitor"),
+    csrfToken: req.csrfToken(),
+    lastAddedEmail,
+    verifiedEmails,
+    unverifiedEmails,
+    countries,
+    usStates,
+    userHasSignedUpForRemoveData,
+    removeData,
+    removeAcctInfo,
+    supportedLocalesIncludesEnglish,
+    whichPartial: partialString,
+    experimentFlags,
+    utmOverrides,
+  });
+}
+
+async function getRemoveConfirmationPage(req, res) {
+  const user = req.user;
+
+  if (!user.kid && !checkIfEnrolledInRemovalPilot(user) && !req.query.show) {
+    return res.redirect("/user/remove-enroll");
+  }
+
+  if (!user.kid && checkIfEnrolledInRemovalPilot(user) && !req.query.show) {
+    return res.redirect("/user/remove-data");
+  }
+
+  const allBreaches = req.app.locals.breaches;
+  const { verifiedEmails, unverifiedEmails } = await getAllEmailsAndBreaches(
+    user,
+    allBreaches
+  );
+  const utmOverrides = getUTMContents(req);
+  const supportedLocalesIncludesEnglish = req.supportedLocales.includes("en");
+  const userHasSignedUpForRemoveData = hasUserSignedUpForWaitlist(
+    user,
+    "remove_data"
+  );
+
+  const experimentFlags = getExperimentFlags(req, EXPERIMENTS_ENABLED);
+
+  let lastAddedEmail = null;
+
+  req.session.user = await DB.setBreachesLastShownNow(user);
+  if (req.session.lastAddedEmail) {
+    lastAddedEmail = req.session.lastAddedEmail;
+    req.session["lastAddedEmail"] = null;
+  }
+
+  let removeData,
```

Comment:
I'm seeing the "remove" namespace is pretty widespread.  Feel free to ignore my comment above if changing to "removal" is too tricky at this point.

Reply:
Yeah - I did it on the controller function names since that was pretty straightforward - but I'll probably pause on everything else.